### PR TITLE
Refactor secretwatcher, storage initializers, and cmd flags

### PR DESCRIPTION
**Pull request by ilackarms**
pretty sizeable PR

refactors the places where secretwatcher as well as secretMap are used (including plugins and flags) so that the new `dependencies.Secret` object and client are used

@yuval-k not ready for review yet, will let you know when (but feel free to TAL if bored)

this PR also removes runtime.HandleError and makes all tests use ginkgowriter (sorry for not moving it to another PR)

### CODE REVIEW

File: test/helpers/config.go [hunk 1]
```diff
@@ -48,12 +47,7 @@ func NewTestUpstream1() *v1.Upstream {
 func NewTestUpstream2() *v1.Upstream {
 	return &v1.Upstream{
 		Name: "localhost-python",
-		Type: service.UpstreamTypeService,
-		Spec: service.EncodeUpstreamSpec(service.UpstreamSpec{
-			Hosts: []service.Host{
-				{Addr: "localhost", Port: 8000},
-			},
-		}),
+		Type: "service",
```
Comment (yuval-k) [inline test/helpers/config.go]: should this be a constant?

File: pkg/bootstrap/secretwatcher/bootstrap.go [hunk 2]
```diff
@@ -55,13 +57,13 @@ func Bootstrap(opts bootstrap.Options) (secretwatcher.Interface, error) {
 				if err != nil {
 					return nil, errors.Wrap(err, "failed to read vault token file")
 				}
-				token = string(b)
+				token = strings.TrimSuffix(string(b), "\n")
```
Comment (yuval-k) [inline pkg/bootstrap/secretwatcher/bootstrap.go]: consider using TrimSpace

File: install/kube/install.yaml [hunk 3]
```diff
@@ -106,6 +106,8 @@ spec:
         - "--storage.refreshrate=1m"
         - "--secrets.type=kube"
         - "--secrets.refreshrate=1m"
+        - "--files.type=kube"
+        - "--files.refreshrate=1m"
```
Comment (yuval-k) [inline install/kube/install.yaml]: will this work on what ever version of gloo that is already there?

File: install/kube/install.yaml [hunk 4]
```diff
@@ -336,7 +336,7 @@ spec:
           value: "1"
         args:
         - "--storage.type=kube"
-        - "--syncperiod=30m"
+        - "--storage.refreshrate=30m"
```
Comment (yuval-k) [inline install/kube/install.yaml]: will this work in the gloo version that's there?

**Comment by yuval-k** [comment]
@ilackarms  overall looks good; some minor comments

File: install/kube/install.yaml (hunk 4 above)
Comment (ilackarms) [reply install/kube/install.yaml]: after we push new images based on these changes.
this is actually a problem.. i would recommend a manual push of images (and updating this PR with those tags) so install.yaml stays consistent

**Pull request merged**